subgraphs/uniswap-forks/protocols/trader-joe/src/common/handlers/handleRewarder.ts

```diff
@@ -0,0 +1,279 @@
+  amount: BigInt;
+  amountUSD: BigDecimal;
+}
+
+class RewardObj {
+  tokenId: string;
+  amount: BigInt;
+  amountUSD: BigDecimal;
+}
+
+export class PoolRewardData {
+  tokens: string[];
+  amounts: Array<BigInt>;
+  amountsUSD: Array<BigDecimal>;
+}
+
+// setPoolRewarder will set, remove or replace a given rewarder
+// when a masterchef staking pool is added or updated. It will not update
+// pool reward tokens, since that will happen on every deposit/withdrawal from the pool.
+export function setPoolRewarder(
+  addr: Address,
+  pool: _MasterChefStakingPool
+): void {
+  if (addr.toHexString() == ZERO_ADDRESS) {
+    removeRewarder(pool);
+    return;
+  }
+
+  replaceRewarder(addr, pool);
+}
+
+// replaceRewarder will remove the current rewarder of a staking pool and add a new one.
+// Token emissions will be updated on the next deposit/withdrawal after the replacement.
+function replaceRewarder(addr: Address, pool: _MasterChefStakingPool): void {
+  removeRewarder(pool);
+  addRewarder(addr, pool);
+}
+
+// removeRewarder will remove the rewarder assigned to some staking pool (if any).
+// Token emissions will be updated on the next deposit/withdrawal after the replacement.
+function removeRewarder(sPool: _MasterChefStakingPool): void {
+  sPool.rewarder = null;
+}
+
+// addRewarder adds a rewarder to a given staking pool and updates the pool
+// reward token and emission values.
+function addRewarder(addr: Address, sPool: _MasterChefStakingPool): void {
+  let rewarder = getOrCreateRewarder(addr, sPool);
+  sPool.rewarder = rewarder.id;
+}
+
+// getPoolRewardsWithBonus will be called on every masterchef deposit or withdrawal
+// to update bonus rewards if a rewarder exists for that pool.
+export function getPoolRewardsWithBonus(
+  event: ethereum.Event,
+  mc: _MasterChef,
+  sPool: _MasterChefStakingPool,
+  liqPool: LiquidityPool,
+  user: Address,
+  isDeposit: boolean = false
+): PoolRewardData | null {
+  if (!sPool.rewarder) {
+    return null;
+  }
+
+  let rewarderAddr = Address.fromString(sPool.rewarder!);
+  let rewarder = getOrCreateRewarder(rewarderAddr, sPool);
+
+  let calc = new RewarderCalculator(mc, rewarder);
+  calc.calculateRewarderRate(event, user, isDeposit);
+
+  return buildRewards(
+    event,
+    liqPool,
+    rewarder,
+    rewarderHasFunds(event, rewarder)
+  );
+}
+
+// rewarderHasFunds will return true if it finds the rewarder has a balance > 0 for the
+// token given as reward. If the last check was too recent, it will return the stored value instead.
+// If it is unable to fetch the balance, it will assume
+// the rewarder DOES have balance.
+function rewarderHasFunds(
+  event: ethereum.Event,
+  rewarder: _MasterChefRewarder
+): boolean {
+  // only check rewarder balance once per day
+  const ONE_DAY = BigInt.fromI32(24 * 3600);
+  if (event.block.timestamp.minus(rewarder.hasFundsAt).lt(ONE_DAY)) {
+    return rewarder.hasFunds;
+  }
+
+  let bal = fetchRewarderBalance(rewarder);
```

Comment:
Already included!

Reply:
What do you mean? 😅

Reply:
My bad, I made this comment before I saw you already include the balancer checker every 24 Hours. Updated the comment above, but forgot this one. Just ignore!